ktor-shared/ktor-serialization/ktor-serialization-gson/jvm/src/GsonConverter.kt

```diff
@@ -36,29 +52,84 @@ public class GsonConverter(private val gson: Gson = Gson()) : ContentConverter {
         return serializeNullable(contentType, charset, typeInfo, value)
     }
 
+    @Suppress("BlockingMethodInNonBlockingContext")
     override suspend fun serializeNullable(
         contentType: ContentType,
         charset: Charset,
         typeInfo: TypeInfo,
         value: Any?
     ): OutgoingContent {
-        return TextContent(gson.toJson(value), contentType.withCharsetIfNeeded(charset))
+        if (!streamRequestBody) {
+            return TextContent(gson.toJson(value), contentType.withCharsetIfNeeded(charset))
+        }
+        return OutputStreamContent(
+            {
+                val writer = this.writer(charset = charset)
+                // specific behavior for kotlinx.coroutines.flow.Flow : emit asynchronous values in Writer
+                if (typeInfo.type == Flow::class) {
+                    (value as Flow<*>).serializeJson(writer)
+                } else {
+                    // non flow content
+                    gson.toJson(value, writer)
+                }
+
+                // must flush manually
+                writer.flush()
+            },
+            contentType.withCharsetIfNeeded(charset)
+        )
     }
 
+    @OptIn(InternalAPI::class)
     override suspend fun deserialize(charset: Charset, typeInfo: TypeInfo, content: ByteReadChannel): Any? {
         if (gson.isExcluded(typeInfo.type)) {
             throw ExcludedTypeGsonException(typeInfo.type)
         }
 
         try {
             return withContext(Dispatchers.IO) {
+                // specific behavior for Sequence : collect it into a List
+                var isSequence = false
```

Comment:
What's the benefit of `call.receive<Sequence<Type>>()` over `call.receive<List<Type>>().asSequence()` . Yes, it's a little bit shorter, but it brings a lot of very unintuitive complexities to the code. Maybe receiving it as a `Flow` would be worth it, to support streamed parsing. But with sequences, there are no benefits, as far as I can see.

Reply:
I think it is ok to use native support where it exists and throw exceptions if the underlying library doesn't support it. Especially, if the alternative is doing some hacks with types.

Reply:
> I think it is ok to use native support where it exists and throw exceptions if the underlying library doesn't support it. Especially, if the alternative is doing some hacks with types.

Noted, I will only keep the Sequence deserialization support in the kotlinx-serialization + UTF-8 case, and remove it everywhere else. This will indeed simplify a lot the code !